rendermodules/point_match_optimization/point_match_optimization.py

```diff
@@ -0,0 +1,313 @@
+    return tempfilename.name
+
+
+def get_tile_pair_matched_image(render, stack, tileId1, tileId2, pGroupId, qGroupId, outdir, matchCollectionOwner, matchCollection, renderScale=0.1):
+    #img1 = renderapi.image.get_tile_image_data(stack, tileId1, scale=renderScale, render=render)
+    #img2 = renderapi.image.get_tile_image_data(stack, tileId2, scale=renderScale, render=render)
+    im1 = '%s.jpg'%(tileId1)
+    im2 = '%s.jpg'%(tileId2)
+    im1 = os.path.join(outdir, im1)
+    im2 = os.path.join(outdir, im2)
+
+    # check if the point match collection exists
+    collections = renderapi.pointmatch.get_matchcollections(owner=matchCollectionOwner, 
+                                                            render=render)
+    ptmatches = []
+    if (matchCollection in collections):
+        ptmatches = renderapi.pointmatch.get_matches_from_tile_to_tile(matchCollection,
+                                                                    pGroupId,
+                                                                    tileId1,
+                                                                    qGroupId,
+                                                                    tileId2,
+                                                                    owner=matchCollectionOwner,
+                                                                    render=render)
+    
+    ptmatch_count = 0
+    if (len(ptmatches) > 0):
+        ptmatch_count = len(ptmatches[0]['matches']['p'][0])
+
+    match_img_filename = draw_matches(im1, im2, ptmatches, renderScale)
+    return match_img_filename, ptmatch_count
+
+
+def compute_point_matches(render, stack, tile1, tile2, pGroupId, qGroupId, output_dir, url_options, option_keys, options):
+    # get canvas urls
+    canvas_urls = get_canvas_url(render.DEFAULT_KWARGS, stack, tile1, tile2, url_options)
+
+    argvs  = []
+    #outdir = 'pmopt'
+    collection_name = 'pms'
+    renderScale = 0.1
+    for param, value in zip(option_keys, options):
+        if param == "renderScale":
+            renderScale = value
+        if type(value) is str:
+            argvs += ['--%s'%(param), '"%s"'%value]
+        else:
+            argvs += ['--%s'%(param), value]
+        #outdir += "_%s_%s"%(param, str(value))
+        collection_name += '_%s'%(str(value).replace('.', 'D'))
+
+    #outdir = os.path.join(output_dir, outdir)
+    outdir = output_dir
+
+
+    baseDataUrl = renderapi.render.format_baseurl(render.DEFAULT_KWARGS['host'], render.DEFAULT_KWARGS['port'])
+    argvs += ['--baseDataUrl', baseDataUrl]
+    argvs += ['--owner', render.DEFAULT_KWARGS['owner']]
+    argvs += ['--collection', collection_name]
+    #argvs += ['--matchStorageFile', os.path.join(outdir, 'matches.json')]
+    argvs += ['--debugDirectory', outdir]
+    argvs += ['', canvas_urls]
+
+    # run the point match client
+    renderapi.client.call_run_ws_client('org.janelia.render.client.PointMatchClient',
```

Comment:
We should add this to render-python's client module.

Reply:
agreed. lets make a render-python issue and we can come back and replace this when that feature is done.

Reply:
i've made an issue